File: ejercicios/02/recursividad.py
def contiene(el, input):
    if el == input:
        return True

    if not isinstance(el, dict):
        return False

    list_of_values = list(el.values())

    if not len(list_of_values) > 0:
        return False

    result = None

    for element in list_of_values:
        result = contiene(element, input)
        if result == True:
            break

    return result


def run():
    objeto_anidado = {
        "data": {
            "informacion": {
                "alguna": {
                    "cosa": {
                        "muy": {
                            "anidada": {
                                "numeroMagico": 33,
                                "otraCosa": "foo2",
                            },
                        },
                    },
                    "otraCosa": {
                        "noTanAnidada": 55,
                    },
                },
            },
        },
    }

    print(contiene(objeto_anidado, 33))  # True
    print(contiene(objeto_anidado, "foo2"))  # True
    print(contiene(objeto_anidado, "foo"))  # False
    print(contiene(objeto_anidado, 55))  # True
    print(contiene(objeto_anidado, 80))  # False


if __name__ == "__main__":
    run()

File: ejercicios/02/recursividad.js
const contiene = (el, input) => {
  if (el === input) return true;
  if (!validation(el)) return false;

  const arrOfValues = Object.values(el);

  // validate if the object is empty
  if (!arrOfValues.length) return false;

  let result;

  for (let i = 0; i < arrOfValues.length; i++) {
    result = contiene(arrOfValues[i], input);
    if (result === true) break;
  }

  return result;
};

const validation = (obj) => typeof obj === "object" && !Array.isArray(obj);

// Tests

const objetoAnidado = {
  data: {
    informacion: {
      alguna: {
        cosa: {
          muy: {
            anidada: {
              numeroMagico: 33,
              otraCosa: "foo2",
            },
          },
        },
        otraCosa: {
          noTanAnidada: 55,
        },
      },
    },
  },
};

console.log(contiene(objetoAnidado, 33)); // true
console.log(contiene(objetoAnidado, "foo2")); // true
console.log(contiene(objetoAnidado, "foo")); // false
console.log(contiene(objetoAnidado, 55)); // true
console.log(contiene(objetoAnidado, 80)); // false
